invisible_cities/detector_simulation/sensor_utils.py

```diff
@@ -0,0 +1,95 @@
+import numpy  as np
+import pandas as pd
+
+from typing import Callable
+from typing import     List
+from typing import    Tuple
+
+from .. database.load_db   import            DataPMT
+from .. database.load_db   import           DataSiPM
+from .. io      .mcinfo_io import get_sensor_binning
+
+
+def trigger_times(trigger_indx: List[int] ,
+                  event_time  :      float,
+                  time_bins   : np.ndarray) -> List[float]:
+    """
+    Calculates trigger time for all identified candidates
+    according to the nexus event time and the time within
+    the event where the trigger was identified.
+
+    Parameters
+    ----------
+    trigger_indx : List[int]
+                   The indices in the histogrammed nexus sensor
+                   data where triggers were identified.
+    event_time   : float
+                   Time of nexus event
+    time_bins    : np.ndarray
+                   The binning used to histogram the nexus sensor
+                   data.
+
+    Returns
+    -------
+    triggered event times : List[float]
+    """
+    return [event_time + time_bins[trg] for trg in trigger_indx]
+
+
+def first_and_last_times(pmt_wfs    : pd.DataFrame,
+                         sipm_wfs   : pd.DataFrame,
+                         pmt_binwid : float       ,
+                         sipm_binwid: float       ) -> Tuple[float, float]:
+    """
+    Returns the maximum and minimum time of an
+    event given the two types of detector.
+    """
+    min_time  = min(pmt_wfs.time.min(), sipm_wfs.time.min())
+    max_time  = max(pmt_wfs.time.max(), sipm_wfs.time.max())
+    max_time += min(pmt_binwid        ,         sipm_binwid)
+    return min_time, max_time
+
+
+def sensor_order(detector_db: str, run_number : int,
+                 n_pmt      : int, length_pmt : int,
+                 n_sipm     : int, length_sipm: int) -> Callable:
+    """
+    Casts the event sensor info into the correct order
+    adding zeros for sensors which didn't see any signal.
+    """
+    pmt_ids    = DataPMT (detector_db, run_number).SensorID
+    sipm_ids   = DataSiPM(detector_db, run_number).SensorID
+    pmt_shape  = (n_pmt , length_pmt )
+    sipm_shape = (n_sipm, length_sipm)
+    def ordering(sensor_order : pd.Int64Index  ,
+                 sensor_resp  : np.ndarray     ,
+                 sensor_shape : Tuple[int, int]) -> np.ndarray:
+        sensors = np.zeros(sensor_shape, np.int)
+        sensors[sensor_order] = sensor_resp
+        return sensors
+        
+    def order_and_pad(pmt_resp    : pd.Series  ,
+                      sipm_resp   : pd.Series  ,
+                      evt_buffers : List[Tuple]) -> List[Tuple]:
```

Comment:
```suggestion
                      evt_buffers : List[Tuple[np.ndarray, np.ndarray]]) -> List[Tuple]:
```
This is getting a bit out of hand.